elasticjob-error-handler/elasticjob-error-handler-dingtalk/src/main/java/org/apache/shardingsphere/elasticjob/error/handler/impl/DingtalkJobErrorHandler.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.shardingsphere.elasticjob.infra.json.GsonFactory;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Job error handler for dingtalk error message.
+ */
+@Slf4j
+public final class DingtalkJobErrorHandler implements JobErrorHandler {
+    
+    private DingtalkConfiguration dingtalkConfiguration;
+    
+    @Override
+    public void handleException(final String jobName, final Throwable cause) {
+        if (dingtalkConfiguration == null) {
+            dingtalkConfiguration = DingtalkEnvironment.getINSTANCE().getDingtalkConfiguration();
+        }
+        HttpURLConnection connection = null;
+        try {
+            URL url = getUrl();
+            connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("POST");
+            connection.setDoOutput(true);
+            connection.setConnectTimeout(dingtalkConfiguration.getConnectTimeout());
+            connection.setReadTimeout(dingtalkConfiguration.getReadTimeout());
+            connection.setRequestProperty("Content-Type", "application/json; charset=utf-8");
+            connection.connect();
+            OutputStream outputStream = connection.getOutputStream();
+            String msg = getMsg(jobName, cause);
+            String paramJson = getParamJson(msg);
+            outputStream.write(paramJson.getBytes(StandardCharsets.UTF_8));
+            int code = connection.getResponseCode();
+            if (HttpURLConnection.HTTP_OK == code) {
+                InputStream resultInputStream = connection.getInputStream();
+                StringBuilder result = new StringBuilder();
+                try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(resultInputStream, StandardCharsets.UTF_8))) {
+                    String line;
+                    while (null != (line = bufferedReader.readLine())) {
+                        result.append(line);
+                    }
+                }
+                JsonObject resp = GsonFactory.getGson().fromJson(result.toString(), JsonObject.class);
+                if (!"0".equals(resp.get("errcode").getAsString())) {
+                    log.error("Job '{}' exception occur in job processing, But the notification Dingtalk failure, error is : {}", jobName, resp.get("errmsg").getAsString(), cause);
```

Comment:
The exception has already <b>"occurred"</b>. I think using past tense is better. 
```
"An exception has occurred in Job '{}'. But failed to send alert by Dingtalk because of {}"
```